certora/specs/Reentrancy.spec

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+methods {
+    function multicall(bytes[]) external returns(bytes[]) => NONDET DELETE;
+
+    function _.accrueInterest(MetaMorphoHarness.MarketParams) external => voidSummary() expect void;
+    function _.supply(MetaMorphoHarness.MarketParams, uint256, uint256, address, bytes) external => uintPairSummary() expect (uint256, uint256);
+    function _.withdraw(MetaMorphoHarness.MarketParams, uint256, uint256, address, address) external => uintPairSummary() expect (uint256, uint256);
+
+    function _.transfer(address, uint256) external => boolSummary() expect bool;
+    function _.transferFrom(address, address, uint256) external => boolSummary() expect bool;
+    function _.balanceOf(address) external => uintSummary() expect uint256;
+}
+
+function voidSummary() {
+    ignoredCall = true;
+}
+
+function uintSummary() returns uint256 {
+    ignoredCall = true;
+    uint256 value;
+    return value;
+}
+
+function uintPairSummary() returns (uint256, uint256) {
+    ignoredCall = true;
+    uint256 firstValue;
+    uint256 secondValue;
+    return (firstValue, secondValue);
+}
+
+function boolSummary() returns bool {
+    ignoredCall = true;
+    bool value;
+    return value;
+}
+
+persistent ghost bool ignoredCall;
+persistent ghost bool hasCall;
```

Comment:
I think that we could have only one "unsafeCall", which is turned true on all calls and turned back to false by the summaries of trusted contracts, but I don't know if it's really better

Reply:
ah no actually the CALL hook is "executed" after the summary?

Reply:
Yes it is executed after the summary
![Screenshot_2024-02-02_11-03-12](https://github.com/morpho-org/metamorpho/assets/22668539/1ce7dc60-9d0b-4da9-9015-7976ac4b54b6)